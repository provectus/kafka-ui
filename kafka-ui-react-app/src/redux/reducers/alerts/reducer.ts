import { getType } from 'typesafe-actions';
import { dismissAlert } from 'redux/actions';
import { Action, AlertsState } from 'redux/interfaces';

import { addError, removeAlert } from './utils';

export const initialState: AlertsState = {};

// eslint-disable-next-line @typescript-eslint/default-param-last
const reducer = (state = initialState, action: Action): AlertsState => {
  const { type } = action;

  if (type.endsWith('__FAILURE')) return addError(state, action);

  if (type === getType(dismissAlert)) {
    return removeAlert(state, action);
  }

  return state;
};

export default reducer;
