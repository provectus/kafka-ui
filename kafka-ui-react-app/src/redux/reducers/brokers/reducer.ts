import { Action, BrokersState, ZooKeeperStatus } from 'redux/interfaces';
import { ClusterStats } from 'generated-sources';

export const initialState: BrokersState = {
  items: [],
  brokerCount: 0,
  zooKeeperStatus: ZooKeeperStatus.offline,
  activeControllers: 0,
  onlinePartitionCount: 0,
  offlinePartitionCount: 0,
  inSyncReplicasCount: 0,
  outOfSyncReplicasCount: 0,
  underReplicatedPartitionCount: 0,
  diskUsage: [],
};

const updateBrokerSegmentSize = (
  state: BrokersState,
  payload: ClusterStats
) => {
  const brokers = state.items;
  const { diskUsage } = payload;

  const items = brokers.map((broker) => {
    const brokerMetrics =
      diskUsage && diskUsage.find(({ brokerId }) => brokerId === broker.id);
    if (brokerMetrics !== undefined) {
      return { ...broker, ...brokerMetrics };
    }
    return broker;
  });

  return { ...state, items, ...payload };
};

const reducer = (state = initialState, action: Action): BrokersState => {
  switch (action.type) {
    case 'GET_BROKERS__REQUEST':
      return initialState;
    case 'GET_BROKERS__SUCCESS':
      return {
        ...state,
        items: action.payload,
      };
    case 'GET_CLUSTER_STATUS__SUCCESS':
      return updateBrokerSegmentSize(state, action.payload);
    default:
      return state;
  }
};

export default reducer;
