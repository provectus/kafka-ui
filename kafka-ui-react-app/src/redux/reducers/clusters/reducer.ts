import { Action } from 'redux/interfaces';
import { Cluster } from 'generated-sources';

export const initialState: Cluster[] = [];

const reducer = (state = initialState, action: Action): Cluster[] => {
  switch (action.type) {
    case 'GET_CLUSTERS__SUCCESS':
      return action.payload;
    default:
      return state;
  }
};

export default reducer;
