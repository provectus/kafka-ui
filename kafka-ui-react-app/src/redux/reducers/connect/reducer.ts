import { getType } from 'typesafe-actions';
import * as actions from 'redux/actions';
import { ConnectState } from 'redux/interfaces/connect';
import { Action } from 'redux/interfaces';

export const initialState: ConnectState = {
  connects: [],
  connectors: [],
};

const reducer = (state = initialState, action: Action): ConnectState => {
  switch (action.type) {
    case getType(actions.fetchConnectsAction.success):
    case getType(actions.fetchConnectorAction.success):
    case getType(actions.fetchConnectorsAction.success):
    case getType(actions.deleteConnectorAction.success):
      return action.payload;
    default:
      return state;
  }
};

export default reducer;
