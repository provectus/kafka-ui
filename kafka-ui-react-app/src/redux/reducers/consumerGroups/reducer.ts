import { Action, ConsumerGroupsState } from 'redux/interfaces';
import { ConsumerGroup } from 'generated-sources';

export const initialState: ConsumerGroupsState = {
  byID: {},
  allIDs: [],
};

const updateConsumerGroupsList = (
  state: ConsumerGroupsState,
  payload: ConsumerGroup[]
): ConsumerGroupsState => {
  const initialMemo: ConsumerGroupsState = {
    ...state,
    allIDs: [],
  };

  return payload.reduce(
    (memo: ConsumerGroupsState, consumerGroup) => ({
      ...memo,
      byID: {
        ...memo.byID,
        [consumerGroup.consumerGroupId]: {
          ...memo.byID[consumerGroup.consumerGroupId],
          ...consumerGroup,
        },
      },
      allIDs: [...memo.allIDs, consumerGroup.consumerGroupId],
    }),
    initialMemo
  );
};

const reducer = (state = initialState, action: Action): ConsumerGroupsState => {
  switch (action.type) {
    case 'GET_CONSUMER_GROUPS__SUCCESS':
      return updateConsumerGroupsList(state, action.payload);
    case 'GET_CONSUMER_GROUP_DETAILS__SUCCESS':
      return {
        ...state,
        byID: {
          ...state.byID,
          [action.payload.consumerGroupID]: {
            ...state.byID[action.payload.consumerGroupID],
            ...action.payload.details,
          },
        },
      };
    default:
      return state;
  }
};

export default reducer;
