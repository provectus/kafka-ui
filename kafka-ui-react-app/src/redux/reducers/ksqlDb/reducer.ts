import { Action } from 'redux/interfaces';
import { getType } from 'typesafe-actions';
import * as actions from 'redux/actions';
import { KsqlState } from 'redux/interfaces/ksqlDb';

export const initialState: KsqlState = {
  streams: [],
  tables: [],
  executionResult: null,
};

// eslint-disable-next-line @typescript-eslint/default-param-last
const reducer = (state = initialState, action: Action): KsqlState => {
  switch (action.type) {
    case getType(actions.fetchKsqlDbTablesAction.success):
      return {
        ...state,
        ...action.payload,
      };
    case getType(actions.executeKsqlAction.success):
      return {
        ...state,
        executionResult: action.payload,
      };
    case getType(actions.resetExecutionResult):
      return {
        ...state,
        executionResult: null,
      };
    default:
      return state;
  }
};

export default reducer;
