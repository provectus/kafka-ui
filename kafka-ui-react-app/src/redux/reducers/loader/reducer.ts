import { Action, LoaderState } from 'redux/interfaces';

export const initialState: LoaderState = {};

const reducer = (state = initialState, action: Action): LoaderState => {
  const { type } = action;
  const matches = /(.*)__(REQUEST|SUCCESS|FAILURE)$/.exec(type);

  // not a *__REQUEST / *__SUCCESS /  *__FAILURE actions, so we ignore them
  if (!matches) return state;

  const [, requestName, requestState] = matches;

  switch (requestState) {
    case 'REQUEST':
      return {
        ...state,
        [requestName]: 'fetching',
      };
    case 'SUCCESS':
      return {
        ...state,
        [requestName]: 'fetched',
      };
    case 'FAILURE':
      return {
        ...state,
        [requestName]: 'errorFetching',
      };
    default:
      return state;
  }
};

export default reducer;
