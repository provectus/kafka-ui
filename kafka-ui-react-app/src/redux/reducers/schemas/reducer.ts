import { SchemaSubject } from 'generated-sources';
import { Action, SchemasState } from 'redux/interfaces';

export const initialState: SchemasState = {
  byName: {},
  allNames: [],
  currentSchemaVersions: [],
};

const updateSchemaList = (
  state: SchemasState,
  payload: SchemaSubject[]
): SchemasState => {
  const initialMemo: SchemasState = {
    ...state,
    allNames: [],
  };

  return payload.reduce((memo: SchemasState, schema) => {
    if (!schema.subject) return memo;
    return {
      ...memo,
      byName: {
        ...memo.byName,
        [schema.subject]: {
          ...memo.byName[schema.subject],
          ...schema,
        },
      },
      allNames: [...memo.allNames, schema.subject],
    };
  }, initialMemo);
};

const reducer = (state = initialState, action: Action): SchemasState => {
  switch (action.type) {
    case 'GET_CLUSTER_SCHEMAS__SUCCESS':
      return updateSchemaList(state, action.payload);
    case 'GET_SCHEMA_VERSIONS__SUCCESS':
      return { ...state, currentSchemaVersions: action.payload };
    default:
      return state;
  }
};

export default reducer;
