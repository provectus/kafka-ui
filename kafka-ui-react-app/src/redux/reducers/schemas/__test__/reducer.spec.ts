import {
  fetchSchemasByClusterNameAction,
  fetchSchemaVersionsAction,
} from 'redux/actions';
import reducer from 'redux/reducers/schemas/reducer';
import {
  clusterSchemasPayload,
  initialState,
  schemaVersionsPayload,
} from './fixtures';

describe('Schemas reducer', () => {
  it('returns the initial state', () => {
    expect(
      reducer(undefined, fetchSchemasByClusterNameAction.request())
    ).toEqual(initialState);
    expect(reducer(undefined, fetchSchemaVersionsAction.request())).toEqual(
      initialState
    );
  });

  it('reacts on GET_CLUSTER_SCHEMAS__SUCCESS and returns payload', () => {
    expect(
      reducer(
        undefined,
        fetchSchemasByClusterNameAction.success(clusterSchemasPayload)
      )
    ).toMatchSnapshot();
  });

  it('reacts on GET_SCHEMA_VERSIONS__SUCCESS and returns payload', () => {
    expect(
      reducer(
        undefined,
        fetchSchemaVersionsAction.success(schemaVersionsPayload)
      )
    ).toMatchSnapshot();
  });
});
