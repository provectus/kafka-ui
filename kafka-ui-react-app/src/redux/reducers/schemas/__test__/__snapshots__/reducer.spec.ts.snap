// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clusters reducer reacts on GET_CLUSTER_SCHEMAS__SUCCESS and returns payload 1`] = `
Object {
  "allNames": Array [
    "test2",
    "test3",
    "test",
  ],
  "byName": Object {
    "test": Object {
      "compatibilityLevel": "BACKWARD",
      "id": 2,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord2\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test",
      "version": "2",
    },
    "test2": Object {
      "compatibilityLevel": "BACKWARD",
      "id": 4,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord4\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test2",
      "version": "3",
    },
    "test3": Object {
      "compatibilityLevel": "BACKWARD",
      "id": 5,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test3",
      "version": "1",
    },
  },
  "currentSchemaVersions": Array [],
}
`;

exports[`Clusters reducer reacts on GET_SCHEMA_VERSIONS__SUCCESS and returns payload 1`] = `
Object {
  "allNames": Array [],
  "byName": Object {},
  "currentSchemaVersions": Array [
    Object {
      "compatibilityLevel": "BACKWARD",
      "id": 1,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord1\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test",
      "version": "1",
    },
    Object {
      "compatibilityLevel": "BACKWARD",
      "id": 2,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord2\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test",
      "version": "2",
    },
  ],
}
`;

exports[`Schemas reducer reacts on GET_CLUSTER_SCHEMAS__SUCCESS and returns payload 1`] = `
Object {
  "allNames": Array [
    "test2",
    "test3",
    "test",
  ],
  "byName": Object {
    "test": Object {
      "compatibilityLevel": "BACKWARD",
      "id": 2,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord2\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test",
      "version": "2",
    },
    "test2": Object {
      "compatibilityLevel": "BACKWARD",
      "id": 4,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord4\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test2",
      "version": "3",
    },
    "test3": Object {
      "compatibilityLevel": "BACKWARD",
      "id": 5,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test3",
      "version": "1",
    },
  },
  "currentSchemaVersions": Array [],
}
`;

exports[`Schemas reducer reacts on GET_SCHEMA_VERSIONS__SUCCESS and returns payload 1`] = `
Object {
  "allNames": Array [],
  "byName": Object {},
  "currentSchemaVersions": Array [
    Object {
      "compatibilityLevel": "BACKWARD",
      "id": 1,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord1\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test",
      "version": "1",
    },
    Object {
      "compatibilityLevel": "BACKWARD",
      "id": 2,
      "schema": "{\\"type\\":\\"record\\",\\"name\\":\\"MyRecord2\\",\\"namespace\\":\\"com.mycompany\\",\\"fields\\":[{\\"name\\":\\"id\\",\\"type\\":\\"long\\"}]}",
      "subject": "test",
      "version": "2",
    },
  ],
}
`;
